iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/VariableDeclarationConverter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.parser;
+
+import java.util.List;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.sonar.iac.arm.tree.api.Identifier;
+import org.sonar.iac.arm.tree.api.PropertyValue;
+import org.sonar.iac.arm.tree.api.VariableDeclaration;
+import org.sonar.iac.arm.tree.impl.json.VariableDeclarationImpl;
+import org.sonar.iac.common.extension.visitors.InputFileContext;
+import org.sonar.iac.common.yaml.tree.MappingTree;
+import org.sonar.iac.common.yaml.tree.TupleTree;
+
+public class VariableDeclarationConverter extends ArmBaseConverter {
+
+  public VariableDeclarationConverter(@Nullable InputFileContext inputFileContext) {
+    super(inputFileContext);
+  }
+
+  public Stream<TupleTree> extractVariablesMapping(MappingTree document) {
+    return document.elements().stream()
+      .filter(filterOnField("variables"))
+      .map(TupleTree::value)
+      .filter(MappingTree.class::isInstance)
+      .map(MappingTree.class::cast)
+      .map(MappingTree::elements)
+      .flatMap(List::stream);
+  }
+
+  public VariableDeclaration convertVariableDeclaration(TupleTree tupleTree) {
+    Identifier name = convertToIdentifier(tupleTree.key());
+    PropertyValue value = convertToPropertyValue(tupleTree.value());
+    return new VariableDeclarationImpl(name, value);
+  }
```

Comment:
https://sonarsource.atlassian.net/browse/SONARIAC-849 Let's improve this in another PR